src/GeoPatchJobs.cpp

```diff
@@ -30,24 +30,32 @@ void SSingleSplitRequest::GenerateMesh() const
 {
 	PROFILE_SCOPED()
 	const int borderedEdgeLen = edgeLen + (BORDER_SIZE * 2);
-#ifndef NDEBUG
 	const int numBorderedVerts = borderedEdgeLen * borderedEdgeLen;
-#endif
 
 	// generate heights plus a 1 unit border
-	double *bhts = borderHeights.get();
-	vector3d *vrts = borderVertexs.get();
+	std::vector<vector3d> positions;
+	positions.reserve(numBorderedVerts);
+	std::vector<double> heightsOut;
+	heightsOut.resize(numBorderedVerts);
 	for (int y = -BORDER_SIZE; y < borderedEdgeLen - BORDER_SIZE; y++) {
 		const double yfrac = double(y) * fracStep;
 		for (int x = -BORDER_SIZE; x < borderedEdgeLen - BORDER_SIZE; x++) {
 			const double xfrac = double(x) * fracStep;
 			const vector3d p = GetSpherePoint(v0, v1, v2, v3, xfrac, yfrac);
-			const double height = pTerrain->GetHeight(p);
-			assert(height >= 0.0f && height <= 1.0f);
-			*(bhts++) = height;
-			*(vrts++) = p * (height + 1.0);
+			positions.emplace_back(p);
 		}
 	}
+	pTerrain->GetHeights(positions, heightsOut);
+
+	double *bhts = borderHeights.get();
+	vector3d *vrts = borderVertexs.get();
+	for (int iHeights = 0; iHeights < numBorderedVerts; iHeights++)
+	{
+		const vector3d &p = positions[iHeights];
+		const double height = heightsOut[iHeights];
+		*(bhts++) = height;
+		*(vrts++) = p * (height + 1.0);
+	}
```

Comment:
Note that instead of allocating an intermediate buffer for positions, you can instead fill the `borderVertexs` and `borderHeights` arrays in the initial pass, and in this pass just modify the `borderVertexs` array like so:
```c++
vrts[i] *= (bhts[i] + 1.0);
```

This results in ~1/6th - ~1/8th less bandwidth in this pass, as you're not copying the heights array from an intermediate buffer.

Reply:
Not going to lie I had to really think about this one for a while! 🤣
I've been looking at and maintaining this code for so many years that this just just never occurred to me! Excellent suggestion!